internal/exec/stages/files/filesystemEntries.go

```diff
@@ -97,6 +97,41 @@ func (s *stage) createCrypttabEntries(config types.Config) error {
 			Source: &uri,
 		})
 	}
+	// if we're creating keyfiles we want to write the containing directories (that don't
+	// already exist) to be mode 0700 rather than auto-creating them at the default directory
+	// permission
+	if len(extrafiles) > 0 {
+		dirMode := 0700
+		dirs := []filesystemEntry{}
+		keyfileDir := distro.LuksRealRootKeyFilePath()
```

Comment:
Instead of iterating up the parents, can we just create that path directly given that this just boils down to `/etc/luks`? (And if an OS isn't shipping `/etc`, I don't think we should create it `0700`).

Reply:
The variable can be overridden via build flag so there's no guarantee that it's always going to fit the case that `/etc/luks` does of only needed the second directory to be created.

Reply:
Ahh gotcha, missed that it was configurable. Hmm, I think it would be safer (and much simpler) to just create the target directory with mode 0700, but OK as is too!

Reply:
A directory node automatically creates all parents with mode 755, which seems like the behavior we want here.  We should still test whether the directory already exists, and create it only if not (because otherwise we'll override its mode), but the loop shouldn't be necessary.

Reply:
That's fair, given the feedback I'll simplify it to just the target keydir. Certainly happy to simplify the code more.